### DISCUSSION TITLE
Nuget package Microsoft.AspNetCore.Authorization.Policy with is not published since 2018

### DISCUSSION THREAD

**Discussion by SwatantraYadav** [dotnet/aspnetcore#26067] [General]
### Describe the bug
Last nuget package published https://www.nuget.org/packages/Microsoft.AspNetCore.Authorization.Policy/2.2.0 does not contain the latest code available on AspNetCore. 

### To Reproduce

1. Try to write code the leverages this code: https://github.com/dotnet/aspnetcore/tree/6c7a8bb397de1fcfeeb6510ea39ef6eb02c15352/src/Security/Authorization/Policy/src
2. Get latest nuget for https://www.nuget.org/packages/Microsoft.AspNetCore.Authorization.Policy
3. Note that none of the code is available

### Further technical details
- ASP.NET Core version 3.1 and 5.0.100-rc.1.20452.10
PS C:\git\AppProtection\src\ManagedCode\Source> dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.402
 Commit:    9b5de826fd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.402\

Host (useful for support):
  Version: 3.1.8
  Commit:  9c1330dedd

.NET Core SDKs installed:
  3.1.202 [C:\Program Files\dotnet\sdk]
  3.1.300 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

Microsoft Visual Studio Enterprise 2019
Version 16.7.3

**Comment by Tratcher** [comment]
A large number of NuGet packages aren't produced for ASP.NET Core 3.0. Such package references should be removed from your project file. This library is now brought in automatically via Microsoft.NET.Sdk.Web or Microsoft.AspNetCore.App.

https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#remove-obsolete-package-references

**Reply by SwatantraYadav** [reply]
Even though my project is targeting Microsoft.NET.Sdk.Web, the Microsoft.AspNetCore.Authorization.Policy package interface is not found. Removing package reference also did not help.

Here is my project configuration: 
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
        <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
    <AssemblyName>Name</AssemblyName>
    <Platforms>x64</Platforms>
  </PropertyGroup>

Severity	Code	Description	
Error	CS0246	The type or namespace name 'IAuthorizationMiddlewareResultHandler' could not be found (are you missing a using directive or an assembly reference?)

**Reply by Tratcher** [reply]
That's a new API introduced in 5.0.0-rc2 which hasn't release yet. https://github.com/dotnet/aspnetcore/pull/21117

**Reply by Tratcher** [reply]
Correction, it's in 5.0.0-preview6 which is publicly available.

**Reply by SwatantraYadav** [reply]
@Tratcher not sure what I am doing wrong here - referencing either of the packages (5.0.0-preview6/5.0.0-rc2) did not help. I still get error "Error CS0246 The type or namespace name 'IAuthorizationMiddlewareResultHandler' could not be found".

Also I could not find the interface mentioned above in Dotnetpeek.

**Reply by Tratcher** [reply]
> referencing either of the packages (5.0.0-preview6/5.0.0-rc2) did not help

What do you mean by that? You should install the latest 5.0 SDK and then have your project target net5.0, you shouldn't need any package references.

**Reply by SwatantraYadav** [reply]
Okay I was trying to target netcoreapp3.1 which would not work. As the Net5.0 is still RC I will wait until its general availability.

**Reply by Tratcher** [reply]
Note the RC is a fully supported "go-live" release.
